packages/@angular/cli/commands/completion.ts

```diff
@@ -117,14 +117,15 @@ const CompletionCommand = Command.extend({
         });
       }
 
-      let opts = '';
+      let opts: string[] = [];
       if (command.blueprints && command.blueprints[0]) {
-        opts += extractBlueprints(command.blueprints);
+        opts = opts.concat(extractBlueprints(command.blueprints));
       }
 
       if (command.availableOptions && command.availableOptions[0]) {
```

Comment:
Oh, interesting, I am glad I learn one bit every day, thanks!

I have no real advise how to properly test it.
Since it is not TypeScript or JavaScript, the test framework does not cover it.
The way I tested it was to try it _manually_ in a bash and in a zsh session, side by side.

Now, with the `. <()` trick, there is no need to remove the `bannerism`.

I would suggest you change the instruction in the banner, replace

```bash
      # Installation: ng completion -b >> ~/.bashrc
      #           or  ng completion -z >> ~/.zshrc
 ```
with
```bash
      # Usage: . <(ng completion --bash) # for bash or
      #              . <(ng completion --zsh) # for zsh
 ```

There is also a small section in Readme.md which would need to be adapted accordingly.